lte/gateway/python/magma/subscriberdb/client.py

```diff
@@ -0,0 +1,167 @@
+        subscriberdb_cloud_client = self._grpc_client_manager.get_client()
+        subscribers = []
+        page_token = ""
+        sync_start = datetime.datetime.now()
+        while True:
+            try:
+                req = ListSubscribersRequest(
+                    page_size=self._subscriber_page_size,
+                    page_token=page_token)
+                response = await grpc_async_wrapper(
+                    subscriberdb_cloud_client.ListSubscribers.future(
+                        req,
+                        self.SUBSCRIBERDB_REQUEST_TIMEOUT
+                    ),
+                    self._loop)
+                subscribers.extend(response.subscribers)
+                if response.next_page_token == "":
+                    logging.info("Successfully fetched all subscriber "
+                                 "pages from the cloud!")
+                    SUBSCRIBER_SYNC_SUCCESS_TOTAL.inc()
+                    time_elapsed = datetime.datetime.now() - sync_start
+                    SUBSCRIBER_SYNC_LATENCY.observe(
+                        time_elapsed.total_seconds() * 1000)
+                    break
+                page_token = response.next_page_token
+            except grpc.RpcError as err:
+                logging.error("Fetch subscribers error! [%s] %s", err.code(),
+                              err.details())
+                time_elapsed = datetime.datetime.now() - sync_start
+                SUBSCRIBER_SYNC_LATENCY.observe(
+                    time_elapsed.total_seconds() * 1000)
+                SUBSCRIBER_SYNC_FAILURE_TOTAL.inc()
+                return None
+
+        return subscribers
+
+    def _process_subscribers(self, subscribers: SubscriberData) -> None:
+        active_subscriber_ids = []
+        sids = []
+        for subscriber in subscribers:
+            sids.append(subscriber.sid.id)
+            if subscriber.lte.state == LTESubscription.ACTIVE:
+                active_subscriber_ids.append(subscriber.sid.id)
+        old_sub_ids = self._store.list_subscribers()
+        # Only compare active subscribers against the database to decide
+        # what to detach.
+        self._detach_deleted_subscribers(old_sub_ids, active_subscriber_ids)
+        logging.debug("Resync with subscribers: %s", ','.join(sids))
+        self._store.resync(subscribers)
+
+    def _detach_deleted_subscribers(self, old_sub_ids, new_sub_ids):
+        """
+        Compares current subscriber ids and new subscriber ids list
+        just streamed from the cloud to figure out the deleted subscribers.
+        Then send grpc DeleteSubscriber request to mme to detach all the
+        deleted subscribers.
+        :param old_sub_ids: a list of old subscriber ids in the store.
+        :param new_sub_ids: a list of new active subscriber ids
+                just streamed from the cloud
+        :return: n/a
+        """
+        # THIS IS A HACK UNTIL WE FIX THIS ON CLOUD
```

Comment:
I think this is fixed now on cloud? Correct me if I'm wrong

https://github.com/magma/magma/blob/95ded6cf88eb6dbd55bfe383cdb98e8fcd7bf6a0/lte/cloud/go/services/policydb/obsidian/models/swagger.v1.yml#L842-L846

Reply:
Any chance we have legacy subscribers that would still be streamed without this?